django/contrib/admin/options.py

```diff
@@ -1505,15 +1506,32 @@ def get_changeform_initial_data(self, request):
         Get the initial form data from the request's GET params.
         """
         initial = dict(request.GET.items())
+        formset_data = defaultdict(lambda: defaultdict(defaultdict))
+        formset_query_regex = r'inlinemode_set-(\d+)-(\d+)-(.*)'
         for k in initial:
             try:
+                match_obj = re.match(formset_query_regex, k)
+                if match_obj:
+                    inline_model_index = int(match_obj.group(1))
+                    inline_model_form_index = int(match_obj.group(2))
+                    inline_model_form_fieldname = match_obj.group(3)
```

Comment:
Note that you can use `match_obj[i]` instead of `match_obj.group(i)`.
See ticket-30116, #12892.

Reply:
I also think that you could shorten these variable names. That would avoid the wrapping below and the backslash line continuation.